DUnitX.Loggers.GUI.VCL.RichEdit.pas

```diff
@@ -0,0 +1,422 @@
+begin
+  //Empty
+end;
+
+procedure TDUnitXGUIVCLRichEditLogger.OnEndTest(const threadId: TThreadID; const Test: ITestResult);
+begin
+  Outdent(1);
+  //WriteBlankLine;
+end;
+
+procedure TDUnitXGUIVCLRichEditLogger.OnEndTestFixture(const threadId: TThreadID; const results: IFixtureResult);
+begin
+  Outdent(3);
+  WriteBlankLine;
+end;
+
+procedure TDUnitXGUIVCLRichEditLogger.OnExecuteTest(const threadId: TThreadID; const Test: ITestInfo);
+begin
+  Write(SExecutingTest + Test.Name);
+  WriteBlankLine;
+end;
+
+procedure TDUnitXGUIVCLRichEditLogger.OnLog(const logType: TLogLevel; const msg: string);
+begin
+  Indent(2);
+  try
+    case logType  of
+      TLogLevel.Information: SetConsoleDefaultColor();
+      TLogLevel.Warning: SetConsoleWarningColor();
+      TLogLevel.Error: SetConsoleErrorColor();
+    end;
+
+    Write(msg);
+  finally
+    Outdent(2);
+    SetConsoleDefaultColor();
+  end;
+end;
+
+procedure TDUnitXGUIVCLRichEditLogger.OnSetupFixture(const threadId: TThreadID; const fixture: ITestFixtureInfo);
+begin
+  Indent(1);
+  Write(SRunningFixtureSetup + fixture.SetupFixtureMethodName);
+end;
+
+procedure TDUnitXGUIVCLRichEditLogger.OnSetupTest(const threadId: TThreadID; const Test: ITestInfo);
+begin
+  SetConsoleSetupTestColor();
+  Write(SRunningSetup + Test.Name);
+end;
+
+procedure TDUnitXGUIVCLRichEditLogger.OnStartTestFixture(const threadId: TThreadID; const fixture: ITestFixtureInfo);
+begin
+  SetConsoleWarningColor();
+  Indent(2);
+  Write(SFixture + fixture.FullName);
+  Write('-------------------------------------------------');
+  Indent(1);
+  SetConsoleDefaultColor();
+end;
+
+procedure TDUnitXGUIVCLRichEditLogger.OnTearDownFixture(const threadId: TThreadID; const fixture: ITestFixtureInfo);
+begin
+  Write(SRunningFixtureTeardown + fixture.TearDownFixtureMethodName);
+  WriteBlankLine;
+end;
+
+procedure TDUnitXGUIVCLRichEditLogger.OnTeardownTest(const threadId: TThreadID; const Test: ITestInfo);
+begin
+  WriteBlankLine;
+  Indent(1);
+  Write(SRunningTestTeardown + test.Name);
+  WriteBlankLine;
+  Outdent(1);
+end;
+
+procedure TDUnitXGUIVCLRichEditLogger.OnTestError(const threadId: TThreadID; const Error: ITestError);
+begin
+  //Empty
+end;
+
+procedure TDUnitXGUIVCLRichEditLogger.OnTestFailure(const threadId: TThreadID; const Failure: ITestError);
+begin
+  //Empty
+end;
+
+procedure TDUnitXGUIVCLRichEditLogger.OnTestIgnored(const threadId: TThreadID; const AIgnored: ITestResult);
+begin
+  //Empty
+end;
+
+procedure TDUnitXGUIVCLRichEditLogger.OnTestingEnds(const RunResults: IRunResults);
+var
+  TestResult: ITestResult;
+begin
+  OutDent(1);
+  Write(SDoneTesting);
+
+  SetConsoleSummaryColor();
+  Write(Format(STestsFound, [RunResults.TestCount]));
+
+  if RunResults.IgnoredCount > 0 then
+    SetConsoleWarningColor()
+  else SetConsoleDefaultColor();
+  Write(Format(STestsIgnored, [RunResults.IgnoredCount]));
```

Comment:
These if...else statements are hard to read when skimming. Might just want linebreaks after the else statements.